internal/testcli/runner.go

```diff
@@ -230,13 +231,60 @@ func (r *Runner) RunBackground() {
 }
 
 func (r *Runner) Run() (bytes.Buffer, bytes.Buffer, error) {
-	r.RunBackground()
-	err := <-r.errch
-	return r.stdout, r.stderr, err
+	r.Helper()
+	var stdout, stderr bytes.Buffer
+	ctx := cmdio.NewContext(r.ctx, &cmdio.Logger{
+		Mode:   flags.ModeAppend,
+		Reader: bufio.Reader{},
+		Writer: &stderr,
+	})
+
+	cli := cmd.New(ctx)
+	cli.SetOut(&stdout)
+	cli.SetErr(&stderr)
+	cli.SetArgs(r.args)
+
+	// Register cleanup function to restore flags to their original values
+	// once test has been executed. This is needed because flag values reside
+	// in a global singleton data-structure, and thus subsequent tests might
+	// otherwise interfere with each other
+	r.registerFlagCleanup(cli)
```

Comment:
This used to be true once in ancient times but is not true anymore. The command objects are not singleton anymore, `cmd.New(ctx)` initializes a completely new command tree. 

We should be able to remove this bit and the `RunBackground` function from here, and the function itself should also be fine to remove.

Reply:
Thanks for this bit of context! I've cleaned up both SetContext and registerFlagCleanup from this PR.

Will clean up it fully in follow up.

Reply:
The integration tests failed after the removal: https://github.com/databricks-eng/eng-dev-ecosystem/actions/runs/12579907431/job/35060907953